plugins/modules/aws_glue_job.py

```diff
@@ -303,11 +345,25 @@ def create_or_update_glue_job(connection, module, glue_job):
         except (BotoCoreError, ClientError) as e:
             module.fail_json_aws(e)
 
+    # handle resource tags
+    resource_arn = _get_glue_job_arn(module, params['Name'])
+
+    new_tags = module.params.get('tags')
+    existing_tags = connection.get_tags(ResourceArn=resource_arn).get('Tags')
+    tags_to_add, tags_to_remove = compare_aws_tags(existing_tags, new_tags if new_tags else {}, module.params.get('purge_tags'))
```

Comment:
If the new tags aren't provided for a task, all are purged now, but before they were just ignored. I'd only add/remove tags if new_tags is not None (`tags: {}` should remove all tags).

Reply:
Ok, maybe my assumption of how `purge_tags: yes` is supposed to work was a bit off.  Is the thought that `purge_tags: yes` would only have an effect if `tags` property was also specified? 

For example:
* if `tags: {}` & `purge_tags: yes` were both specified, this would result in removal of all existing tags?
* if just `purge_tags: yes` was specified, this would result in no action being taken on the existing tags?